src/components/Web3SendTransactionButton.tsx

```diff
@@ -1,54 +1,73 @@
 import { Button, CircularProgress, Tooltip } from "@material-ui/core";
 import DoneIcon from "@material-ui/icons/Done";
 import ErrorIcon from "@material-ui/icons/Error";
+import TimerIcon from "@material-ui/icons/Timer";
+import moment from "moment";
 import React, { useState } from "react";
 import { Transaction } from "web3-core/types";
+import { now } from "../time";
 import MetamaskIcon from "./MetamaskIcon";
 import NoWeb3Tooltip from "./NoWeb3Tooltip";
 import { useWeb3 } from "./Web3Context";
 
 interface Props {
   transaction: Transaction;
-  actionReady: boolean;
+  minTimestamp?: number;
 }
 
 enum TransactionState {
   Initial,
   Signing,
   Sent,
-  Error
+  Error,
+  Unavailable
 }
 
-function Web3SendTransactionButton({ transaction, actionReady }: Props) {
+function Web3SendTransactionButton({ transaction, minTimestamp }: Props) {
   const web3 = useWeb3();
   const [state, setState] = useState(TransactionState.Initial);
 
+  const whenReadyMessage =
```

Comment:
:x: I think we should ditch this completely. Comparing current time to the block timestamps will never be accurate.

Reply:
I think it's probably accurate enough for the purposes of a message, but if you have a very strong opinion about this...

Reply:
It looks good with your suggestion anyway, so not a bad change.